keanu-project/src/main/java/io/improbable/keanu/plating/loop/LoopBuilder.java

```diff
@@ -0,0 +1,180 @@
+package io.improbable.keanu.plating.loop;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+
+import io.improbable.keanu.plating.Plate;
+import io.improbable.keanu.plating.PlateBuilder;
+import io.improbable.keanu.plating.Plates;
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.VertexLabelException;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.BoolProxyVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.DoubleProxyVertex;
+import io.improbable.keanu.vertices.generic.nonprobabilistic.If;
+
+public class LoopBuilder {
+    private final Collection<Vertex> initialState;
+    private ImmutableMap.Builder<VertexLabel, VertexLabel> customMappings = ImmutableMap.builder();
+    private int maxLoopCount = Loop.DEFAULT_MAX_COUNT;
+    private boolean throwWhenMaxCountIsReached = true;
+
+    <V extends Vertex<?>> LoopBuilder(Collection<V> initialState) {
+        this.initialState = ImmutableList.copyOf(initialState);
+    }
+
+    /**
+     * An optional method to override the default value
+     *
+     * @param maxCount - the max number of times the loop can run
+     * @return self
+     */
+    public LoopBuilder atMost(int maxCount) {
```

Comment:
The name here should be more descriptive I think. `withMaxIterations` or `atMostIterations`...

Reply:
done (withMaxIterations)